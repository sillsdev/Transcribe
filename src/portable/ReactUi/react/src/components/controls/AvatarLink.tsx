import * as React from 'react';
import Avatar from 'react-avatar';
import { Link } from 'react-router-dom';
import './AvatarLink.sass';

interface IProps {
    id: string;
    name: string;
    select?: (id: string) => any;
    size?: string;
    target: string;
    uri: string;
    avatarShape?: boolean
};

class AvatarLink extends React.Component<IProps, object> {
    public render() {
        const { avatarShape, id, name, select=(() => undefined), size="64", target, uri } = this.props;
        return (
            <div id={id} className="AvatarLink">
                <Link to={target} onClick={select.bind(this, id)}>
                    <Avatar className="OnHover"  name={name} src={uri} size={size} round={(avatarShape !== undefined && !avatarShape)? avatarShape: true}/>
                    <br /> <br />
                    <div className="caption">{name}</div>
                </Link>
            </div>
        )
    }
};

export default AvatarLink;
