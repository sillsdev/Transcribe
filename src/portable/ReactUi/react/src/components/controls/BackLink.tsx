import * as React from 'react';
import { Link } from 'react-router-dom';
import './BackLink.sass';

interface IProps {
    action?: (context: any) => any;
    target: string;
    disable?:boolean;
};

class BackLink extends React.Component<IProps, object> {
    public render() {
        const { action, target, disable=false } = this.props;
        const linkWrapper = disable? "\u276E":(<Link to={target} onClick={action}>{"\u276E"}</Link>);
        return (
            <div id="BackLink" className="BackLink">
                {linkWrapper}
            </div>
        )
    }
};

export default BackLink;
