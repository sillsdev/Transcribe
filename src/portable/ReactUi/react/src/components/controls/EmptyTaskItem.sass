@import "../../_variables.sass"

.EmptyTaskItem 
	width: 336px
	height: 192px
	background: no-repeat url("/assets/EmptyTaskIcon.png")
	display: flex
	flex-direction: column-reverse
	padding-bottom: 16px
	background: linear-gradient(180deg, #073642 0%, rgba(7, 54, 66, 0) 100%)
	mix-blend-mode: overlay
	opacity: 0.8
	transform: rotate(180deg)
	&>*
		display: flex
		flex: 1 0 auto
		flex-direction: column
		background: linear-gradient($base02, rgba(7, 54, 66, 0))

	.ImageStyle
		display: flex
		flex-direction: column

	.CaptionStyle
		display: flex
		flex-direction: column
		transform: inherit
		padding: 0px 10px
		font-family: Roboto
		font-style: normal
		font-weight: normal
		line-height: 28px
		font-size: 16px
		letter-spacing: 0.444444px
		.labelNone
			color: $base3

		.labelText
			color: $base1