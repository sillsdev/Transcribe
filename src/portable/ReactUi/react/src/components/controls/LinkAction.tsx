import * as React from 'react';
import { Button } from 'react-bootstrap'
import './LinkAction.sass';

interface IProps {
    target:  (event: React.MouseEvent<Button>) => any;
    text: string;
};

class LinkAction extends React.Component<IProps, object> {
    // public linkClick() {
    //     // tslint:disable-next-line:no-console
    //     console.log("Save button clicked!!");
    // }
    public render() {
        const { target, text } = this.props;
        return (
            <div id="LinkAction" className="LinkAction">
                <Button
                    onClick={target}
                    bsStyle="link"
                    bsSize="small">
                    {text}
                </Button>
            </div>
        )
    }
};

export default LinkAction;
