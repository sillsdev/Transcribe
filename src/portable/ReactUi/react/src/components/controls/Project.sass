@import "../../_variables.sass"

.Project
	width: 180px
	height: 180px
	text-align: initial
	flex: 0 0 auto
	.main
		width: inherit
		height: inherit
		position: absolute
		-moz-border-width: 2px
		border-width: 2px
		border: solid transparent
		background-color: rgba($taskBG,0.5)
	.main .adminDiv
		display: none
		width: inherit
		height: inherit
	.main:hover
		-moz-border-width: 2px
		border-width: 2px
		-moz-border-top-colors: $cyan2
		-moz-border-bottom-colors: $cyan2
		-moz-border-right-colors: $cyan2
		-moz-border-left-colors: $cyan2
		border: solid $cyan2
		box-sizing: border-box
		-moz-box-sizing: border-box
		background-color: rgba($taskBG,0.5)
	.main:hover .adminDiv
		display: block
	.mainNotAdmin
		-moz-border-width: 2px
		border-width: 2px
		border: solid transparent
		cursor: default
		background-color: rgba($taskBG,0.5)
		width: inherit
		height: inherit
		position: absolute
	.captionDiv
		position: absolute
		width: inherit
		height: inherit
		display: flex
		flex-direction: row
		justify-content: center
	.caption
		color: $litUpText
		font-family: $sans
		font-size: 34px
		line-height: 40px
		letter-spacing: 0.25px
		overflow: hidden
		text-overflow: clip
		white-space: nowrap
		padding: 20px 0
	.projectImage
		opacity: 0.5
		width: inherit
		height: inherit
		position: absolute
	.adminIcon
		position: absolute
		display: flex
		align-items: center
		justify-content: center
		top: 77.13px
		width: inherit
		height: 70px
		margin: auto
	.adminCaption
		position: absolute
		text-align: center
		color: $litUpText
		font-family: $sans
		font-size: 14px
		line-height: 16px
		letter-spacing: 0.75px
		width: inherit
		height: 17px
		top: 147px
	.ContextMenu
		position: absolute
		display: flex
		color: $litUpText
		line-height: 16px
		width: 170px
		height: 16px
		top: 8px
		justify-content: flex-end
	nav.react-contextmenu
		background-color: $navBG
		text-align: left
		color: $base3
		padding: 16px
	.react-contextmenu-item
		width: auto
		height: 32px
		font-family: $sans
		font-size: 16px

	a:active
		outline: none