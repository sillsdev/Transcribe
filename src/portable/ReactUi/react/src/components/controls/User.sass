@import "../../_variables.sass"

.User
	padding: 20px 0 20px 0
	width: 180px
	height: 180px
	text-align: center
	display: block
	.caption
		text-align: center
		color: $litUpText
		font-family: $sans
		font-size: 16px
		line-height: 28px
		letter-spacing: 0.44px
	.role
		text-align: center
		color: $litUpText
		font-family: $sans
		font-size: 12px
		line-height: 16px
		letter-spacing: 0.4px
	.main
		padding-top: 20px
		padding-bottom: 20px
		-moz-border-width: 2px
		border-width: 2px
		border: solid transparent
	.main:hover
		-moz-border-width: 2px
		border-width: 2px
		-moz-border-top-colors: $cyan2
		-moz-border-bottom-colors: $cyan2
		-moz-border-right-colors: $cyan2
		-moz-border-left-colors: $cyan2
		border: solid $cyan2
		box-sizing: border-box
		-moz-box-sizing: border-box
		background-color: $taskBG
	a:active
		outline: none
