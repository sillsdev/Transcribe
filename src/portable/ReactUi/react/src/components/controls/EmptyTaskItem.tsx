import * as React from 'react';
import './EmptyTaskItem.sass';

interface IProps {
    id: string;
};

class EmptyTaskItem extends React.Component<IProps, object> {
    public render() {
        const { id } = this.props;
        return (
            <div id={id} className={"EmptyTaskItem"}>
                <div className="ImageStyle">
                    <img src={"/assets/EmptyTaskIcon.svg"} />
                </div>
                <div className="CaptionStyle">
                    <label className="labelNone">None</label>
                </div>
            </div>
        )
    }
};

export default EmptyTaskItem;
