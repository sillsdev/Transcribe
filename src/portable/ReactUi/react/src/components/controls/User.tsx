import * as React from 'react';
import Avatar from 'react-avatar';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { IProjectSettingsStrings } from '../../model/localize';
import { IState } from '../../model/state';
import userStrings from '../../selectors/localize';
import './User.sass';

interface IProps extends IStateProps {
    id: string;
    name: string;
    select?: (id: string) => any;
    size?: string;
    target: string;
    uri: string;
    role: string[];
};

export class User extends React.Component<IProps, object> {
    public render() {
        const { id, name, select=(() => undefined), size="64", target, uri, role, strings } = this.props;
        let workingAs
        if (role && role.indexOf("administrator") >= 0) {
            workingAs = strings.admin
        } else if (role.indexOf("reviewer") >= 0) {
            workingAs = strings.reviewer
        } else {
            workingAs = strings.transcriber
        }
        return (
                <Link id={id} to={target} onClick={select.bind(this, id)} className="User">
                    <div className="main">
                        <Avatar name={name} src={uri} size={size} round={true}/>
                        <br /> <br />
                        <div className="caption">{name}</div>
                        <div className="role">{workingAs}</div>
                    </div>
                </Link>
        )
    }
};

interface IStateProps {
    strings: IProjectSettingsStrings;
};

const mapStateToProps = (state: IState): IStateProps => ({
    strings: userStrings(state, {layout: "projectSettings"}),
});

export default connect(mapStateToProps)(User);
