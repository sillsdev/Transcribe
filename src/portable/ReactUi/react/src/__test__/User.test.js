/* jshint esversion: 6 */
import React from 'react';
import expect from 'expect';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import {User} from '../components/controls/User';

configure({ adapter: new Adapter() })

// Snapshot for User
describe('>>>Control: User --- Snapshot',()=>{
    const minProps = {
        id: "my id",
        name: "my name",
        role: ["my role1","my role2", "my role3"],
        strings: {
            admin: "Admin",
            reviewer: "Reviewer",
            transcriber: "Transcriber"
        },
        target: "/root",
        uri: "images/myImage.png"
    };

    it('+++capturing Snapshot of User', () => {
        const renderedValue =  renderer.create(<Router><User {...minProps}/></Router>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
//*******************************************************************************************************
describe('>>>Control: User', () => {
    let wrapper;
    const minProps = {
        id: "my id",
        name: "my name",
        role: ["my role1","my role2", "my role3"],
        strings: {
            admin: "Admin",
            reviewer: "Reviewer",
            transcriber: "Transcriber"
        },
        target: "/root",
        uri: "images/myImage.png",
    };

    beforeEach(() => {
        wrapper = shallow(<User {...minProps} />);
    })

    it('+++ renders without exploding', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('+++ activates select link action', () => {
        const onButtonClick = sinon.spy();
        const linkWrapper = shallow(<User {...minProps} select={onButtonClick}/>)
        linkWrapper.find('Link').simulate('click');
        expect(onButtonClick.calledOnce).toEqual(true);
    });
});
