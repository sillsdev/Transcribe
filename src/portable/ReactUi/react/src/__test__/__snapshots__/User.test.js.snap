// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Control: User --- Snapshot +++capturing Snapshot of User 1`] = `
<a
  className="User"
  href="/root"
  id="my id"
  onClick={[Function]}
>
  <div
    className="main"
  >
    <div
      className=" sb-avatar"
      onClick={undefined}
      style={
        Object {
          "borderRadius": "100%",
          "display": "inline-block",
          "fontFamily": "Helvetica, Arial, sans-serif",
          "height": "64px",
          "verticalAlign": "middle",
          "width": "64px",
        }
      }
    >
      <div
        className=" sb-avatar__text"
        style={
          Object {
            "background": undefined,
            "borderRadius": "100%",
            "color": "#FFF",
            "height": "64px",
            "lineHeight": "initial",
            "textAlign": "center",
            "textTransform": "uppercase",
            "width": "64px",
          }
        }
      >
        <div
          style={
            Object {
              "display": "table",
              "height": "100%",
              "tableLayout": "fixed",
              "width": "100%",
            }
          }
        >
          <span
            style={
              Object {
                "display": "table-cell",
                "fontSize": "100%",
                "verticalAlign": "middle",
                "whiteSpace": "nowrap",
              }
            }
          >
            <span />
          </span>
        </div>
      </div>
    </div>
    <br />
     
    <br />
    <div
      className="caption"
    >
      my name
    </div>
    <div
      className="role"
    >
      Transcriber
    </div>
  </div>
</a>
`;
